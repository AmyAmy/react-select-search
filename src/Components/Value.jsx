import React, { memo } from 'react';
import PropTypes from 'prop-types';

const Value = ({
    search,
    placeholder,
    multiple,
    snapshot,
    cls,
    autoFocus,
    autoComplete,
    renderValue,
    valueProps,
}) => {
    const inputValue = (snapshot.focus && search) ? snapshot.search : snapshot.displayValue;
    const shouldRender = (!multiple || placeholder) || search;

    if (!shouldRender) {
        return null;
    }

    return (
        <div className={cls('value')}>
            {renderValue(
                {
                    ...valueProps,
                    placeholder,
                    autoFocus,
                    autoComplete,
                    value: inputValue,
                },
                snapshot,
                cls('input'),
            )}
        </div>
    );
};

Value.defaultProps = {
    placeholder: null,
    renderValue: (valueProps, snapshot, className) => (
        <input
            {...valueProps}
            className={className}
        />
    ),
};

Value.propTypes = {
    search: PropTypes.bool.isRequired,
    multiple: PropTypes.bool.isRequired,
    autoFocus: PropTypes.bool.isRequired,
    autoComplete: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    valueProps: PropTypes.shape({
        tabIndex: PropTypes.string,
        readOnly: PropTypes.bool,
        onFocus: PropTypes.func,
        onBlur: PropTypes.func,
        onKeyPress: PropTypes.func,
        onKeyDown: PropTypes.func,
        onKeyUp: PropTypes.func,
        onChange: PropTypes.func,
        disabled: PropTypes.bool,
        ref: PropTypes.oneOfType([
            PropTypes.func,
            PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
        ]),
    }).isRequired,
    snapshot: PropTypes.shape({
        focus: PropTypes.bool.isRequired,
        search: PropTypes.string.isRequired,
        displayValue: PropTypes.string.isRequired,
    }).isRequired,
    cls: PropTypes.func.isRequired,
    renderValue: PropTypes.func,
};

export default memo(Value);
