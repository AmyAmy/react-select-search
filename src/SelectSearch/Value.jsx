import React from 'react';
import PropTypes from 'prop-types';

const Value = props => (
    <input
        className="select-search-box__search"
        {...props}
    />
);

Value.defaultProps = {
    placeholder: '',
    onChange: null,
    type: 'text',
};

Value.propTypes = {
    tabIndex: PropTypes.string.isRequired,
    onFocus: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
    readOnly: PropTypes.bool.isRequired,
    value: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    onChange: PropTypes.func,
    type: PropTypes.string,
};

export default Value;
