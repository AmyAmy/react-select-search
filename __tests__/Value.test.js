import renderer from 'react-test-renderer';
import Value from '../src/Components/Value';

describe('Test Value component', () => {
    test('Single, without search', () => {
        const component = renderer.create(
            <Value
                search={false}
                multiple={false}
                autoFocus={false}
                placeholder=""
                valueProps={{
                    tabIndex: '0',
                    readOnly: true,
                }}
                snapshot={{
                    focus: false,
                    search: false,
                    displayValue: 'Hamburger',
                }}
                cls={(cls) => cls}
            />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        expect(component.root.findByType('input').props.value).toBe('Hamburger');
    });

    test('Single, with search', () => {
        const component = renderer.create(
            <Value
                search={true}
                multiple={false}
                autoFocus={false}
                placeholder=""
                valueProps={{
                    tabIndex: '0',
                    readOnly: false,
                }}
                snapshot={{
                    focus: true,
                    search: 'ham',
                    displayValue: 'Hamburger',
                }}
                cls={(cls) => cls}
            />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        expect(component.root.findByType('input').props.value).toBe('ham');
    });
});
