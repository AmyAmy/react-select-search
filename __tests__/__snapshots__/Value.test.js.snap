// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Value component Single, with search 1`] = `
<div
  className="value"
>
  <input
    autoFocus={false}
    className="input"
    placeholder=""
    readOnly={false}
    tabIndex="0"
    value="ham"
  />
</div>
`;

exports[`Test Value component Single, without search 1`] = `
<div
  className="value"
>
  <input
    autoFocus={false}
    className="input"
    placeholder=""
    readOnly={true}
    tabIndex="0"
    value="Hamburger"
  />
</div>
`;
